### PR TITLE
Chore: Updating prometheus to 0.51.2

### PR DESCRIPTION
**Pull request by pedro-stanaka**
## Summary

Just a small update to Prometheus 0.51.2.

### CODE REVIEW

File: engine/engine.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 	"github.com/prometheus/prometheus/storage"
 	"github.com/prometheus/prometheus/util/annotations"
 	"github.com/prometheus/prometheus/util/stats"
+	v1 "github.com/prometheus/prometheus/web/api/v1"
```
Comment (fpetkovski) [inline engine/engine.go:23]: Hmm was this interface moved back?

Comment (pedro-stanaka) [reply engine/engine.go:23]: Related PR: https://github.com/prometheus/prometheus/pull/13713

Comment (MichaHoffmann) [reply engine/engine.go:23]: Should we perspectively move fallback into the consumers of this library? i.e. make thanos use prometheus engine if creation of a query fails with thanos engine? That way we can skip fallback and prometheus engine here.

**Review by MichaHoffmann — APPROVED** [review]
lgtm

File: engine/engine.go (hunk 1 above)
Comment (pedro-stanaka) [reply engine/engine.go]: nvm, you were right. I was not taking `main`.

**Comment by pedro-stanaka** [comment]
Seems like upstream prometheus changed behavior of some aggregation functions, I will try to work on this to unblock the update moving forward. I will have to touch in the implementation here as well. https://github.com/prometheus/prometheus/pull/13744

**Comment by MichaHoffmann** [comment]
The tests are only failing if the first aggregation parameter is a series right?

**Comment by pedro-stanaka** [comment]
> The tests are only failing if the first aggregation parameter is a series right?

Yes

**Comment by MichaHoffmann** [comment]
I think it's a big in Prometheus

**Pull request closed without merging**